htdocs/lib2/logic/user.class.php

```diff
@@ -1586,26 +1586,25 @@ public function getGivenRatings()
         );
     }
 
-    public function getMaxRatings()
+    public function getRatingParameters()
     {
         global $opt;
 
-        // get number of possible rates
-        return (int) floor($this->getStatFound() * $opt['logic']['rating']['percentageOfFounds'] / 100);
-    }
+        $findsPerRating = $opt['logic']['rating']['findsPerRating'];
+        $finds = $this->getStatFound();
+        $ratings = $this->getGivenRatings();
 
-    public function allowRatings()
-    {
-        // new ratings allowed, if "given ratings" < "max ratings"
-        return ($this->getGivenRatings() < $this->getMaxRatings());
+        return [
+            'maxRatings' => floor($finds / $findsPerRating),
+            'givenRatings' => $ratings,
+            'findsUntilNextRating' => ($ratings + 1) * $findsPerRating - $finds,
```

Comment:
fixed two bugs in calculation of findsUntilNextRating; inverted logic of `$opt['logic']['rating']['findsPerRating']` avoids the possibility of non-integer values